x-pack/plugins/features/server/ui_capabilities_for_features.test.ts

```diff
@@ -135,6 +135,42 @@ describe('populateUICapabilities', () => {
     });
   });
 
+  it(`supports capabilities from reserved privileges`, () => {
```

Comment:
Minor nit: you've got a test case below for `supports merging features with sub privileges`, do you think it would be worthwhile to add something like `supports merging features with sub privileges and reserved privileges` too?

Reply:
That's actually not a case that we support right now. It might technically work here, but you can only register sub-feature privileges if you declare non-reserved privileges, and we don't support registering both reserved and non-reserved privileges at the same time